docs/docs/dev_docs/contracts/syntax/historical_trees.md

```diff
@@ -0,0 +1,227 @@
+
+Instead of passing the nullifier, you can check that a note has not been nullified by passing the note.
+
+TODO #include_code prove_note_not_nullified yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
+
+## Public value inclusion
+
+This proves that a public value exists at a certain block.
+
+### 1. Import public_value_inclusion
+
+```rust
+aztec::{
+history::{
+    #include_code import_public_value_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr raw
+}
+}
+```
+
+### 2. Call prove_public_value_inclusion
+
+`prove_public_value_inclusion` takes 3 parameters:
+
+| Name            | Type                   | Description                                         |
+|-----------------|------------------------|-----------------------------------------------------|
+| value | Field                   | The public value you are proving inclusion for             |
+| storage_slot    | Field                    | Storage slot the value exists in          |
+| block_number         | u32         | Block number for proving value's existence     |
+| context         | PrivateContext         | Private context     |
+
+### Example
+
+#include_code prove_public_value_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
+
+## Contract inclusion
+
+This proves that a contract exists in, ie had been deployed before or in, a certain block.
+
+### 1. Import contract_inclusion
+
+```rust
+aztec::{
+history::{
+    #include_code import_contract_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr raw
+}
+}
+```
+
+### 2. Call prove_contract_inclusion
+
+`prove_contract_inclusion` takes 7 parameters:
+
+| Name                      | Type            | Description                                           |
+|---------------------------|-----------------|-------------------------------------------------------|
+| deployer_public_key       | GrumpkinPoint   | Public key of the contract deployer                   |
+| contract_address_salt     | Field           | Unique identifier for the contract's address          |
+| function_tree_root        | Field           | Root of the contract's function tree                  |
+| constructor_hash          | Field           | Hash of the contract's constructor                    |
+| portal_contract_address   | EthAddress      | Ethereum address of the associated portal contract             |
+| block_number              | u32             | Block number for proof verification                   |
+| context                   | PrivateContext  | Private context                    |
+
+If there is no associated portal contract, you can use a zero Ethereum address:
+
+```ts
+new EthAddress(Buffer.alloc(EthAddress.SIZE_IN_BYTES));
+```
+
+### Example
+
+TODO #include_code prove_contract_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
```

Comment:
Can this TODO be removed?

Also update to whole function? This function has a lot of comments above it, those could be cut out of the reference.

https://github.com/AztecProtocol/aztec-packages/blob/2ac1b9c82ece6edbb3e91e998c2cb5693277e8d9/yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr#L204-L231